### ISSUE TITLE
2.2 Release progress

### ISSUE THREAD

**Issue by markelog** [jquery/jquery#2723]
Instead of porting bug fixes to `2.2-stable` branch i decided to go with opposite way - reverting back-incompat commits.

I tried more conventional approach at first (and another three too), but it proved (to me) to be ineffective. Since there was a lot more conflicts then with chosen approach which were prompted by the use of the new QUnit API and style fixes.

Below, you find three lists, one of the reverted commits and one of the questionable commits that are in `2.2-stable` and are back-compat, but might be inappropriate for this release. And also list of reverted commits which requires special attention.

I would expect from reviewers to check those lists and current state of the branch, verifying that API doesn't include new features, don't break back-compat and done overall correctly.

Processed commits started from Oct 30, 2014.

### Questionable commits

Drop support for Opera 12.x, Firefox<29, Safari<6.0 and old Blackberry
https://github.com/jquery/jquery/commit/740e190223d19a114d5373758127285d14d6b71e

Callbacks module change, considered as a bug
https://github.com/jquery/jquery/commit/89ce0af2cf7f001647e74fc1de92ce94a51fd5c2

Hides `jQuery.swap`
https://github.com/jquery/jquery/commit/bb4d888f4f7886371347b59eae5d4e6135acb7ed

Adds `jQuery.htmlPrefilter` - allowed by the semver for the minor version,
but might be inappropriate for this release, also not documented.
https://github.com/jquery/jquery/commit/225bde37c997f5ddd9fe00fdfb8e9a43545cfbbc

Changes signature of the `jQuery#init`, needed for the Migrate (/cc @dmethvin)
https://github.com/jquery/jquery/commit/7a6931de689865f559bed26e797e8cbb5674bbce

Adds `jQuery.uniqueSort`, deprecates `jQuery.unique` - allowed by the semver for the minor version, but might be not contracdict 2.2 release, also not documented.
https://github.com/jquery/jquery/commit/e1090c3d2b2a988a5b41f1f1ed9f8d6dcae02200

Hides `jQuery.domManip`
https://github.com/jquery/jquery/commit/62d5579578109f1468a37e44f76af06f283a46ab

Hides `jQuery.buildFragment`
https://github.com/jquery/jquery/commit/a74320fca8bb0a2190f6e1fdb71a73733b6986e4

Changes signature (remove additional param) of `jQuery.event.simulate` couldn't find docs for it, if documented is not acceptable change for the minor
https://github.com/jquery/jquery/commit/3655c4e1908cc3ee49487e7e26e8cfca9fe8146d

Hides `jQuery.acceptData`
https://github.com/jquery/jquery/commit/224271982eb9cd351d7db1b38c740b4e927e6f97

Hides `jQuery.dir` and `jQuery.sibling`
https://github.com/jquery/jquery/commit/f9ef427d355a8a2b531aed5d112dfa4f9692105c

Even though subject of the commit reads "Adding unit tests for jQuery.Animation"
There was changes in logic and interface of the `effects` module (/cc @gnarf)
https://github.com/jquery/jquery/commit/b3b2d6c3dd51fbdc69e1942e9af75cc99a1834c2

### Reverted commits

Reason: API change (behaviour)
Name: Ajax: Always use script injection in globalEval
Commit: https://github.com/jquery/jquery/commit/bbdfbb4ee859fe9319b348d88120ddc2c9efbd63
Issue: trac-14757
PR: https://github.com/jquery/jquery/pull/1449

Reason: API change (behaviour)
Name: Effects: Reintroduce use of requestAnimationFrame
Commit: https://github.com/jquery/jquery/commit/72119e0023dcc0d9807caf6d988598b74abdc937
Issue: trac-15147
PR: https://github.com/jquery/jquery/pull/1578

Reason: Follow-up commit for
Name: Effects: Improve raf logic
Commit: https://github.com/jquery/jquery/commit/708764f47b0c8de152bbb444d0f608db558b76ed
PR: https://github.com/jquery/jquery/pull/1578

Reason: API change (removed methods)
Name: Deprecated: Drop size and andSelf methods
Commit: https://github.com/jquery/jquery/commit/f110360f65a268e959ae892ca36e85da3d91e606
Issue: https://github.com/jquery/jquery/issues/1749
PR: https://github.com/jquery/jquery/pull/1822

Reason: Adherence to the latest version of DOM specification
Name: Manipulation: support data-URI scripts insertion
Commit: https://github.com/jquery/jquery/commit/15f4dec7894f1e00adbfb9bce4f870441a527bd6
Issue: https://github.com/jquery/jquery/issues/1887
PR: https://github.com/jquery/jquery/pull/1888

Reason: API change (behaviour)
Name: Core: Throw an error on $("#") rather than returning 0-length collection
Commit: https://github.com/jquery/jquery/commit/80022c81ce4a07a232afd3c580b0977555a2daec
Issue: https://github.com/jquery/jquery/pull/1682

Reason: Follow-up commit for "Adherence to the latest version of DOM specification"
Name: Manipulation: improve test for data-URI
Commit: https://github.com/jquery/jquery/commit/dd596ccf729b2f39d44b73bc54b53bd41c880146

Reason: API change (behaviour)
Name: Manipulation: make wrapAll funarg execute only once
Commit: https://github.com/jquery/jquery/commit/359b03cac74d7336676a6992f14b7ccab9b28659
Issue: https://github.com/jquery/jquery/issues/1843
PR: https://github.com/jquery/jquery/pull/1912

Reason: API change (behaviour)
Name: Callbacks: Don't abort execution on .lock()
Commit: https://github.com/jquery/jquery/commit/32bf9178cace8c52a31c468a28adde9a4fdc4671
Issue: https://github.com/jquery/jquery/issues/1990
PR: https://github.com/jquery/jquery/pull/1991

Reason: API change (proprties removal)
Name: Core: Remove deprecated context and selector properties
Commit: https://github.com/jquery/jquery/commit/0ea8c32863af31fb5cfc184e8d513bbae35583e8
Issue: https://github.com/jquery/jquery/issues/1908
PR: https://github.com/jquery/jquery/pull/2000

Reason: API change (return value)
Name: Core: Return empty array instead of null for parseHTML("")
Commit: https://github.com/jquery/jquery/commit/4116914dcac32868c2c822366785e3dd2ccc69d4
Issue: https://github.com/jquery/jquery/issues/1997
PR: https://github.com/jquery/jquery/pull/1998

Reason: API change (return value)
Name: Follow-up commit for "Manipulation: support data-URI scripts insertion"
Commit: https://github.com/jquery/jquery/commit/4fae91141c04dffb5094b5961eb5d7b2a1f3f423
Issue: https://github.com/jquery/jquery/issues/1993

Reason: API change (methods removal)
Name: Ajax: remove deprecated extensions from ajax promise
Commit: https://github.com/jquery/jquery/commit/9d1b989f20b550af3590691723b0620f6914626e
Issue: https://github.com/jquery/jquery/issues/2084
PR: https://github.com/jquery/jquery/pull/2092

Reason: Promise interoperability
Name: Deferred: Backwards-compatible standards interoperability
Commit: https://github.com/jquery/jquery/commit/555a50d340706e3e1e0de09231050493d0ad841e
Issue: https://github.com/jquery/jquery/issues/1722
PR: https://github.com/jquery/jquery/pull/1996

Reason: API change (feature)
Name: Wrap: Support .unwrap( selector) for selective unwrapping
Commit: https://github.com/jquery/jquery/commit/7b09235ceed57bbcc26fc2c76147eb4e95ebdb92
Issue: https://github.com/jquery/jquery/issues/1744
PR: https://github.com/jquery/jquery/pull/2003
Note: Could be included in the minor release, need documentation

Reason: API change (behaviour)
Name: Event: remove guard for falsy handler argument of jQuery#on method
Commit: https://github.com/jquery/jquery/commit/fac67a984268ef8f7de952666fda6d8d32754f5f
Issue: https://github.com/jquery/jquery/issues/2248
PR: https://github.com/jquery/jquery/pull/2249

Reason: API change (behaviour)
Name: Data: do not include digits when camelCasing
Commit: https://github.com/jquery/jquery/commit/2862a07af640be42711f72ad4abe532349985824
Issue: https://github.com/jquery/jquery/issues/1751

Reason: API change (behaviour)
Name: Data: always camelCase keys in .data()
Commit: https://github.com/jquery/jquery/commit/0e790985a76fd813a6e56696c87abeed5a6cf63b
Issue: https://github.com/jquery/jquery/issues/2257

Reason: API change (behaviour)
Name: Data: camelCasing should not ignore case
Commit: https://github.com/jquery/jquery/commit/172cad80ac635d8900aa6a3504c89f38b320488e
Issue: https://github.com/jquery/jquery/issues/2070

Reason: API change (behaviour)
Name: Offset: Fix .offset() to correctly work with ShadowDOM
Commit: https://github.com/jquery/jquery/commit/1617479fcf7cbdaf33dc9334ed10a0f30bf14687
Issue: https://github.com/jquery/jquery/issues/1784
PR: https://github.com/jquery/jquery/pull/2043

Reason: Follow-up commit for "Offset: Fix .offset() to correctly work with ShadowDOM"
Name: Offset: allow offset setter to throw for disconnected elements
Commit: https://github.com/jquery/jquery/commit/0d11c1182f2012cd6eb06ce1e3fa5a495af9bee3
Issue: https://github.com/jquery/jquery/issues/2114

Reason: Follow-up commit for "Offset: Fix .offset() to correctly work with ShadowDOM"
Name: Offset: remove ownerDocument check in offset getter
Commit: https://github.com/jquery/jquery/commit/6176567361e9d05d4f5eb7e98735a678f2cd7ea9
Issue: https://github.com/jquery/jquery/issues/2115

Reason: API change (behaviour)
Name: CSS: Ignore the CSS cascade in show()/hide()/etc.
Commit: https://github.com/jquery/jquery/commit/86419b10bfa5e3b71a7d416288ab806d47a31d1f
Issue: https://github.com/jquery/jquery/issues/1767
Issue: https://github.com/jquery/jquery/issues/2071
PR: https://github.com/jquery/jquery/pull/2180

Reason: Follow-up commit for "CSS: Ignore the CSS cascade in show()/hide()/etc."
Name: CSS: Make show/hide/toggle methods a module
Commit: https://github.com/jquery/jquery/commit/67d7a2eefee768b59eb3d51cb1fb2c671873e58a
Issue: https://github.com/jquery/jquery/issues/2193
PR: https://github.com/jquery/jquery/pull/2648

Reason: Follow-up commit for "CSS: Ignore the CSS cascade in show()/hide()/etc."
Name: CSS: fix AMD mode for the new showHide module
Commit: https://github.com/jquery/jquery/commit/32cfc38a9c76d1f9163759c811cb9c82eb47d565

Reason: Follow-up commit for "CSS: Ignore the CSS cascade in show()/hide()/etc."  --
Name: Effects: fix loading showHide in AMD mode
Commit: https://github.com/jquery/jquery/commit/ab06be561ec74cccaa2d581830210f82326f05c3

Reason: API change (event removal)
Name: Core: remove custom ready event
Commit: https://github.com/jquery/jquery/commit/c252c5fac2a25869524c2eba2e93d4230e6af25c
Issue: https://github.com/jquery/jquery/issues/2264
PR: https://github.com/jquery/jquery/pull/2265

Reason: API change (events removal)
Name: Event: remove deprecated event aliases
Commit: https://github.com/jquery/jquery/commit/0705be475092aede1eddae01319ec931fb9c65fc
Issue: https://github.com/jquery/jquery/issues/2286
PR: https://github.com/jquery/jquery/pull/2287

Reason: Follow-up commit for "Event: remove deprecated event aliases"
Name: Ajax: Remove remnants of the load event alias handling
Commit: https://github.com/jquery/jquery/commit/38a669735d08bcbd28cfb0d77eee82c67aa89eeb
PR: https://github.com/jquery/jquery/pull/2362

Reason: Revert of the API change
Name: Revert "Offset: allow offset setter to throw for disconnected elements"
Commit: https://github.com/jquery/jquery/commit/578dcee96a8d4d759b3a7e623177fa36a5133ba7

Reason: API change (new method)
Name: Deferred: add .catch handler
Commit: https://github.com/jquery/jquery/commit/84ccf2606c6b97d5875774bf774f9f2aae950ae7
Issue: https://github.com/jquery/jquery/issues/2102
Note: Could be included in the minor release, need documentation

Reason: API change (behaviour)
Name: CSS: Make .css("width") & .css("height") return fractional values
Commit: https://github.com/jquery/jquery/commit/b60b26e18477d21fa5ec9c6572e114fc5d441730
Issue: https://github.com/jquery/jquery/issues/1724
PR: https://github.com/jquery/jquery/pull/2439

Reason: API change (signature)
Name: Remove additional parameters of easings
Commit: https://github.com/jquery/jquery/commit/b7a7dea95f84d6d8e5a8186d4fb09a762baf79bb
Issue: https://github.com/jquery/jquery/issues/2367
PR: https://github.com/jquery/jquery/pull/2376

Reason: API change (behaviour)
Name: Ajax: Mitigate possible XSS vulnerability
Commit: https://github.com/jquery/jquery/commit/b078a62013782c7424a4a61a240c23c4c0b42614
Issue: https://github.com/jquery/jquery/issues/2432
PR: https://github.com/jquery/jquery/pull/2588

Reason: Follow-up commit for "Ajax: Mitigate possible XSS vulnerability"
Name: Ajax: don't expect cross-origin tests run in envs which not support it
Commit: https://github.com/jquery/jquery/commit/39cdb8c9aa0fde68f733553ba050a2ba9d86474c

Reason: API change (behaviour)
Name: Attributes: do not set properties to false when removing booleans
Commit: https://github.com/jquery/jquery/commit/47ccf3daadc4b312f850502300129952e70f9d9d
Issue: https://github.com/jquery/jquery/issues/1759

Reason: Follow-up commit for "Attributes: do not set properties to false when removing booleans"
Name: Attributes: remove flakey test for selected attribute
Commit: https://github.com/jquery/jquery/commit/87bd130289c6ed9bfc355c1f8587ae6ce00a4776

Reason: API change (behaviour)
Name: Ajax: Only form-encode requests with a bod
Commit: https://github.com/jquery/jquery/commit/70605c8e5655da996ebd395e3c43423daaa08d9c
Issue: https://github.com/jquery/jquery/issues/2658
PR: https://github.com/jquery/jquery/pull/2671

Reason: API change (behaviour)
Name: Attributes: return empty array for select-multiple with no values
Commit: https://github.com/jquery/jquery/commit/79fc806e8500372a2278795c068d039ee287535f
Issue: https://github.com/jquery/jquery/issues/2562
PR: https://github.com/jquery/jquery/pull/2689

Reason: API change (behaviour)
Name: Dimensions: outerWidth/Height include scrollbar
Commit: https://github.com/jquery/jquery/commit/7d44d7f9e7cb73ff2b373f08cea13ea9958bb462
Issue: https://github.com/jquery/jquery/issues/1729
PR: https://github.com/jquery/jquery/pull/2694

Reason: API change (return value)
Name: Dimensions: Empty sets should return undefined
Commit: https://github.com/jquery/jquery/commit/97d79ecf6b6c5cc3c1485eb3c46e12986a978f57

Reason: API change (signature change)
Name: Manipulation: execute scripts from iframe in the iframe's context
Commit: https://github.com/jquery/jquery/commit/22449eb968622c2e14d6c8d8de2cf1e1ba4adccd
Issue: https://github.com/jquery/jquery/issues/1757
PR: https://github.com/jquery/jquery/pull/2696
Note: depends on "Ajax: Always use script injection in globalEval"

### Complicated reverts

Compatibility with standard Promises (/cc @gibson042)
https://github.com/jquery/jquery/commit/2d9d1f006ac98b3dd1f23de1b721f54eb77346db

`show()`/`hide()` changes (/cc @gibson042)
https://github.com/jquery/jquery/commit/0169459d5d896a87e4cc0026bfd9b823d5525220
https://github.com/jquery/jquery/commit/87db099ed54d0dac624c4ae6b476328e88aef2d7
https://github.com/jquery/jquery/commit/88df7676d735b4cf16dd56ac55fbdef3319f66e7
https://github.com/jquery/jquery/commit/43996d166ee72bc5180edeca2b3312d5c40709cd

--

You might notice that i included new `jQuery#htmlPrefilter` but didn't `jQuery.Deferred#catch`.
Why? Because i think it might be a good opportunity to test new methods in the wild (would require a documentation updates though), but not the ones that are related to module rewrites (like `catch()`).

If this method would be found productive it could be applied to 1.x branch.

/cc @jquery/core

**Comment by dmethvin** [comment]
Wow. Thanks @markelog I know this must have been a ton of work!

**Comment by timmywil** [comment]
Great work!

**Comment by mzgol** [comment]
## The things IMO we should *not do* in 1.12/2.2:
> Drop support for Opera 12.x, Firefox<29, Safari<6.0 and old Blackberry
> 740e190

IMO dropping support for some browsers is a breaking change and I don't think we should do it before 3.0.0.

> Even though subject of the commit reads "Adding unit tests for jQuery.Animation"
> There was changes in logic and interface of the effects module (/cc @gnarf)
> b3b2d6c

These tests are problematic, they break a lot in Android 2.3. I'd rather not add animation/tween tests in 1.12/2.2

## The things that IMO we may do in 1.12/2.2:
> Callbacks module change, considered as a bug
> 89ce0af

I think it's a new feature, not a bug fix. But we're allowed to introduce non-breaking new features in a minor release so that's fine.

> Hides jQuery.swap
> bb4d888

It was undocumented so that's fine.

> Adds jQuery.htmlPrefilter - allowed by the semver for the minor version,
> but might be not contracdict 2.2 release, also not documented.
> 225bde3

I don't understand the "but might be not contracdict 2.2 release" part. But the change looks OK to me.

> Changes signature of the jQuery#init, needed for the Migrate (/cc @dmethvin)
> 7a6931d

`jQuery#init` is undocumented so we may do whatever we want with it, especially in a minor.

## Other remarks
> You might notice that i included new jQuery#htmlPrefilter but didn't jQuery.Deferred#catch.
> Why? Because i think it might be a good opportunity to test new methods in the wild (would require a documentation updates though), but not the ones that are related to module rewrites (like catch()).

I'm very happy you didn't include `Deferred#catch`. Some Promises/A+-compatible promise libraries special-case jQuery by checking for `.pipe` and reject them as they're not Promises/A+-compliant. The fact that from 3.0.0 Deferreds will have `catch` allows those libs to check for it and allow such jQuery promises. Otherwise it will be very hard to support.

I'm trying to do sth like that in this [PR for `chai-as-promised`](https://github.com/domenic/chai-as-promised/pull/122).

**Comment by timmywil** [comment]
This still needs verification by the team. Any and all that can review this week, please do.

**Comment by gibson042** [comment]
I agree with @mzgol that 740e190 (drop support) should be reverted in 1.x/2.x so we maintain support levels. Is there anything you want from me regarding 2d9d1f0 and show/hide changes?

**Comment by mzgol** [comment]
I've already reverted the commit that dropped support for some browsers on
the 2.2-stable branch. :)

-- 
Michał Gołębiowski

**Comment by markelog** [comment]
Yeah, that was decided one meeting ago, we would need to make opera 12.17 to pass now, is anyone working on that?

Also, any other concerns? Remember, that it is preferable to check as much as possible. As decided on the meeting, deadline is on wednesday.

**Comment by mzgol** [comment]
I'll try to check the commits but I can't promise I'll be able to do it
before Thursday; we'll see. Remaining 2 Opera bugs can be fixed afterwards,
they don't seem to me major (unless I'm wrong :)).

**Comment by markelog** [comment]
> Remaining 2 Opera bugs can be fixed afterwards,

This could became an avalanche of reverts. 

@timmywil i think we can move deadline to thursday then?

**Comment by mzgol** [comment]
>> Remaining 2 Opera bugs can be fixed afterwards,
>
> This could became an avalanche of reverts. 

I'd expect to not revert anything else but just fix the bugs. I really don't think this is so important to necessarily get fixed within the next 3 days.

One is [`onerror` handler not called in `jQuery.parseHTML`](http://swarm.jquery.org/result/818496), maybe Opera is lacking an API or sth.

Another one is [`.css('lineHeight', '+=50%')` not working](http://swarm.jquery.org/result/818490).

**Comment by dmethvin** [comment]
if the regressions are minor but the code changes would be difficult, I'd prefer just noting the regressions in the release notes. We're talking a pretty old version of Opera after all.

**Comment by mzgol** [comment]
> if the regressions are minor but the code changes would be difficult, I'd prefer just noting the regressions in the release notes. We're talking a pretty old version of Opera after all.

I can look at it but definitely not before Thursday. ;) Otherwise, I agree, I just want to make sure there's no easy fix for those issues.

**Comment by markelog** [comment]
So did anyone had a chance to look at it more closely?

**Comment by markelog** [comment]
Okay, waiting until next meeting then

**Comment by markelog** [comment]
Friendly ping

**Comment by timmywil** [comment]
I did 2 things. I used `git cherry` to check which commits were added to 2.2-stable and then diffed jquery.js from 2.1.4 and jquery.js from 2.2-stable, to double-check I wasn't missing any significant code changes.

This is the list I generated from git cherry (with all reverted commits removed). These are the commits that have been added on top of 2.1.4...

+ 72119e0023dcc0d9807caf6d988598b74abdc937 Effects: Reintroduce use of requestAnimationFrame
+ 708764f47b0c8de152bbb444d0f608db558b76ed Effects: Improve raf logic
+ fbdbb6fd431a3f598b581079b0ca37c399d369d4 Build: Move test to appropriate module
+ 39b76064d9400295a383291ab6e821e259b67832 Build: Update commitplease dev dependency
+ 323e82c09f9078cc16f0df170e7a8bddea33523e Build: Upgrade to grunt-bowercopy 1.0.0
+ 8d113104e980179e9eb4d092df0b1ab37cdc6fa5 Build: Remove unused Sizzle test files
+ c5d9d88dce65b89a1d7d12b41ad254d5017a92a8 Build: Move all external libraries to external directory
+ 35f8e15fd73256ec02397d7bf93893e38afc9bc9 Build: ignore test dependencies for npm install
+ c0b23e2b71eb7dd19b6435c6e94980275dd3990d Build: update Sizzle to 1.11.1 and include license
+ 712e78c47fc3e550dabbb44055e5bf7877871bff Build: update grunt-bowercopy
+ 4f776e5ff91c48891f02d1ca3ba31132477d3972 Build: Update license
+ 86e62d8b37ff9ad40e7c21c2c0c440a9cdfc550e Selector: Remove "#" exception for identifier tokens
+ bcca4f041b4c98dc9cb127ea3a2a7f40d2db4796 Build: update Sizzle to 2.0.0
+ 2fb719e5aa885232c5347576e9f9e1e341a7ba15 Core: Consistently use local reference to access()
+ df62159296e9de8b04653ba999a32f9be7bb2a73 Readme: Fix punctuation in tile
- e5190982c40d7ac8ab9bdb2e7e4334f0e123ef66 Ajax: Remove workaround for IE6/7
+ e1949f43711b5808d42378bafb6728b40b1738d6 Build: drop bower; use npm for front-end deps
+ 8356948ed4ee13af218af74c56c8a91ee9523828 Build: update front-end dependencies
+ 8e3a0ceafa2c7c78902d0eab07d21b793deb5366 Build: update node dependencies barring jscs
+ c869a1ef8a031342e817a2c063179a787ff57239 Build: update grunt-jscs-checker and pass with the new rules
+ 269a27c70204c7d233eac3cd91a383e9b5759a2f Build: update source map options for the new grunt jshint
+ 10399ddcf8a239acc27bdec9231b996b178224d3 CSS: elements are hidden when either offsetWidth or offsetHeight is zero
+ 995f70777ac6c0f988a44807ef1399e73937b2ee Css: Fix the "sanity check" test
+ 895ea6887dd02a44067d041db00c355c80e930a3 CSS: Removed redundant "to the number" in comment
+ 29838b6cab6f2e508f3e9692f32918c72b1a504b Core: Drop strundefined variable
+ 4287442ab8bdf8d9f008f3e84ccbf7d86d0fb5db Offset: Simplified a conditional
+ 15a609f7663c4348ab7f1acbc9e566ec20bb717c Build: Moved JSHint directives to .jshintrc file
+ 075da3091cda170bd8cd5ce47bad1c5b14760232 Attr: Use typeof check for getAttribute method
+ 274feb53cc9a99633dfac785d8b3b837d192c43c CSS: Remove use of getDefaultComputedStyle
+ a7724186c98734529b06bebb8f2dc7fc2b641011 CSS: Use pre-defined displays for html and body
+ 7e09619cdf2813f2cd9da600ba682be8f91b08b6 CSS: Correct typo in the comment
+ 609adf63dadb7052c816b1a3ee44e3f928ede2e3 CONTRIBUTING: Close parenthesis
+ 1ae025e24f9920d18cec8a8498bfc4eed7e3f1dc Manipulation: Tolerate XMLNode host object input to getAll
+ b807aedb7fee321fb3aa5d156a5a256ab0634e2d Event: Restore the `constructor` property on jQuery.Event prototype
- 0cdec797de23555c95a70978f4d9e06f3b041330 Data: avoid non-alphanumeric chars in expando properties
+ 2e9c1ead76aaae8fde004e3cacde1e36c5fd2104 README: Improve build instructions
+ edfc94df92982c2d840f6692291e09211d1c7cfc Readme: Fix minor style issues. Thanks @MightyBranch!
- 4e7f34f6296111f7f91d621397dfb02c6bf4c41f Ajax: remove event dependency from the ajax module
+ ff928f5775566ee54f26fcb0f9a326719c32e63c Misc: Need for speed removed by 9ad6e7e
+ 3e89a53265bfdc992d7a97a01c5d1025e48e5906 Misc: Update all references to bugs.jquery.com
+ 5bc1ddc5cc96e509e0fadb0c1e3a70927de6ed1c Build: Upgrade to commitplease 2.0.0
+ 1d752731011d35e985271334ff98186728949f03 README: Update the description of the deprecated module
+ 31d58c5cbbede6dc33d10feecd5277ccddb367e6 Change broken url to wayback one
+ 598ed050f6931922214aeaea8a1dc867e0cc6fb7 Ajax: Fix for request aborted in ajaxSend
+ 7a607c5673fa36f9818f19e349871600a2938b1e Build: update version to 3.0.0-pre
+ 758fd6cea9e82f7bfebce07ba6ecf0d107e8a53c Docs: 1.x-master branch -> compat branch; 2.x branch -> master branch
+ e81b258ace257ba138022cf28374190623b0be5e Misc: Remove leftover -moz-box-sizing in tests
+ c309b95756b83b60fadc369edb120db4520c364a Misc: Adjust comments & docs to dropping IE<8 in jQuery Compat
+ 1bbb678949da480dce0d9ed3d0fb273bccce9787 Manipulation: Check state lost if the name is set for Android 4.0-4.3
+ dc4b914a05e1e6dbf13f916897b5d6a472ea0380 Build: Fix various typos
+ 9d6beac3958da79671344fd11b9a3fe9b85f88e1 Docs: correct grunt command in README.md
+ f7e60dc83d81cbf892de9dab39642dd67c49bd23 Dimensions: allow modification of coordinates argument
- 7d15b4d1f181de8868c375c545a51e4dfc28a611 CSS: Clean up memory leak in reliableMarginRight
+ cfe2eae38df411c0f15cbbf927612bc515063bf5 README: Add selector-native.js link. Thanks @randsonjs!
+ d9ed166c865e91ccc8cef1ca282785c500ca2306 Event: Copy detail property to jQuery.Event on native events
+ bfd5dab2c6f387dce08e7c153ad0b3854ed971e4 CONTRIBUTING: Condense info and add directions to other resources
+ 2c1b556d98da597b0490f204e3561f656987f17c Build: Remove empty define({}) from build output
+ 6748ba349650353c7bed6eec201a3192f6b2cae1 Tests: Update QUnit
+ 26276a307ce2f554b63a05ed8451155f01711c29 Tests: Minor updates for QUnit 1.16 compatibility
+ 3c13f4c6297566a71102c2362347987f6d6a636e Tests: Accommodate page changes from the QUnit HTML reporter
+ 906caebb3f3bc542904a94846e778aa8d71c0575 Tests: Tilt at a few style guide windmills
+ bc1cb122db30be034b3da84ee2546c86e2a5268f Callbacks: Disabling a callback should prevent firing
+ 4467ed606ab0a9b28ed495e682576910937fa5e0 Event: Fully clean up events in unit test
+ f6302b0b53d61dfe1adbfaf6612be5cbced5bbc1 Attributes: Use the option val hook in select val hook and simplify it
+ eeda11cdd6a69ec2ef4e9c1ed12b82a79910fda5 Core: Simplify and speed up .each
+ 43faf6d1f922ba44a84c93f4ff2461d208b2bf48 Support: Re-organize browser order, add Safari 8
+ 58c24608210c9a9a264a38746628ebc26823f59b Core: use document.implemenation.createHTMLDocument in jQuery.parseHTML
+ 6051609df35ef5e478c79c76534c03e4b46100bf Build: fix tests in AMD mode
+ 59232825aa87b941dd2418a6860b64017dfec0ae Core: remove unnecessary support test for createHTMLDocument
+ 31c7d7fb7530e1af950b41d13dd956820d5c6908 Core: pass empty string to createHTMLDocument to appease IE
+ 4ab743188ec6bdaafe8550dab84f374ef7e22aca Css: Remove non-functional unit test for negative margin
+ b77983173e18724a883e02ad3a84661e18e6cf4a Core: revert addition of createHTMLDocument. Thanks, Safari 8.
+ 8653068dd6b8a515f5c1d8a0fda4479e9534103e Event: Empty namespaces should be uneventfully ignored
+ cfe468f29c4cbe1a457d0feb17dec90dcfd7c280 Core: re-introduce createHTMLDocument in parseHTML; Safari 8 left out
+ b091fdbafac33123cba329e6bb48b9281323ca38 Ajax: use anchor tag for parsing urls
+ ce308e25e57a0a040cd1ea05f00bf1cc23c1bd8a Support: Correct iOS 8 support test results, re-arrange entries
+ 31b63fc23674ede657f019783ad7b942bb83f9c3 README: various text fixes
+ 7602dc708dc6d9d0ae9982aadb9fa4615a9c49fa Sizzle: update to 2.1.1
+ 03557db96c5b9fad22320e94b52b118c89f7b6f9 Authors: Update AUTHORS.TXT and .mailmap
+ 78ac753192d5490295a46b60f278c030d958bf6d Build: account for version labels in Sizzle versions
+ 91017047fc85cf634d9c7726901c250708f58731 Build: update node dependencies
+ d6c97abb744bfe8c67ab9158aecdb5bb1c05e47b Build: Sizzle version labels must start with a dash
+ d21edb599d8f5f80a3f3ecba5b62311b9cd1a3e6 Event: fix incorrect window bug with scrollTop/Left in iframes
+ 1b932bb7867924b98529537f8ccd35db979ae22a CSS: Add unit tests for negative margins and positioning
+ 72c4a067a9def9720a997f23615690970894dc9c Data: speed up $.fn.data() for camel-cased key
+ 6f65f5faf2b5831414d4c93d271a59265b41a75b Data: shave off a couple of bytes
+ c62486fb4cb18fdb7dc5807231c964ed82ee6482 CSS: simplify "defaultDisplay" module
+ 4ab8603669e2bbc8644e402927c33ce422d7aaa3 Offset: don't run scrollTop/scrollLeft iframe test in mobile Safari
+ 5a0867d1e94794fbfc3be1f18f4c0a168dc18d95 Ajax: simplify jQuery.parseXML method
+ ab20d9d24f0a95df382c02e9ef3dcc4adb86e4f1 Deferred: Fix $.when with resolved deferred and progress callbacks
+ 76df9e4e389d80bff410a9e5f08b848de1d21a2f Build: Don't assume the browser environment; smoke test on Node w/ jsdom
+ b410b15e8da1a8d285f431640ff165e634196115 README: update the Homebrew site address
+ 66e1b6b8d49812239b5712d65922ff94c60f7b02 Build: Remove dates from copyright notice
+ 97cf5280824027c3d4fcdbb4db49c10ad3c62bce Event: Normalize mouse event properties in drag events
+ d30c482910bf35c5dbaec9c82174d1e6ae8839b5 Misc: Mac OS is now OS X, thanks @xfq
+ 83b038fc81ec12e29dba92ad022e1d5fcf4980bc Ajax: Fix cross-domain detection test for non-default port
+ c34ed46eee12c92cb050a53eb62daf9e1258caf9 Release: Remove copying of jquery-latest files
+ a3779bc3254159f5a99e8204e69a0f0b9e191f6f Docs: Clarify custom build instructions
+ b50e0f2c3651d64d40931a16733b015f56409c63 Docs: Fix README uppercase
+ 0c9d018889944da3d60cac135dc38dfcc37ac14a Build: Specify valid components for commit messages
+ 0d829f0e9a177038580a59d0b5649fab76b8104f Callbacks: No object starts out locked
+ 53aa87f3bf4284763405f3eb8affff296e55ba4f Core: Standardize indexOf comparisons
+ bb928bde7e7b85357fef3fedd450b04c03e965d7 Build: Rearrange grunt/npm tasks into a build/dist/test pattern
+ 89ce0af2cf7f001647e74fc1de92ce94a51fd5c2 Ajax: $.post and $.get can now take an options object
+ a4133ffafa6ac43b3aa22bc4edb4444b09f5250c Build: Sanctify the component name status of Wrap
+ 534f13025aeb59a7d5893ce62efafa0b066a4934 Core: add support to tag-hyphenated elements
+ 85ffc6d973865a031ded170934e0acfc2e97cb11 Manipulation: add support to tag-hyphenated elements
+ 74ae5444832b2fb966768a97281d2ad8c088bc58 Offset: simplify jQuery#offsetParent method
+ d7e5fcee519e5f3e840beef9e67a536e75133df9 Event: HTML5 drop events inherit from MouseEvent
+ e905dcd8f33e99275e29b0333e7b255559197c81 Release: update AUTHORS.txt
+ a5e1c9b44c971fd7046d9a95bd0810e50840b663 Event: Add a note about a mouseenter bug in Chrome
+ 31f4f8e3f30e67e677a2aa167b9e894d46d5b81e Build: Speed up the Travis build
+ 087d280ad1160de53a45ea96184911194f7b46e0 Data: Use a PDF object instead of a Java applet for acceptData testing
+ 26eca143c2dd857b9e3d1c446a467fed16e903d2 Release: Distribute files to distribution repo
+ 61224f5fef4336cd47cb26be74f9758096f7f956 Release: bower.json is actually generated from scratch
+ aae998b5449f35dc71e87e360aee5bd0022e7c8e Release: dist can be run during a test
+ e0673dfedb9ad07d8e68f28a54453b975c412c33 Release: fix CDN archive creation
+ a76c7812366e7e45ea29969db72d90261ef87af5 Release: remove sourcemap comment from all copies of minified file
+ 1ba45fcc15c894cad591d93cbb88010df5f235fe Release: push dist to same remote as project
+ 3747cc642a48d2a5a8ac83069f66bddd33bea301 CSS: Restore the hack to get pixels for .css('width') etc.
+ cdfc2d092afad5a3e5b3978b04b402a1ee9dce79 CSS: Fix the pixelMarginRight support test in Android 2.3
+ 45ec73f55618cd080867a98b42b9ab80409cab2e CSS: save 20 bytes in css/support
+ 0ea342a6a6dce793c1b0f14f051c2573f40f4e44 Manipulation: simplify html wrappers
+ e984d1c79cc476062818e03df04a366baa13d197 Manipulation: don't auto-insert tbody
+ 0c466438d1feaec302fec13eafeeb063bbfd6a19 Offset: don't run scrollTop/scrollLeft iframe test in Android 2.3 & 4.0
+ 2d5c5d213f09fa0205d07a2d60a36581058cc40a Build: Update QUnit to latest (1.17.1)
+ fcb6c4d1d6552c7e54df16a36b171858bdf0553f Core: simplify "each" stylesheet iteration test
+ 0ac28ed293681cb8f2e9fdd11efa0021da039c84 Ajax: simplify one ajax call and add explanatory comment
+ 4ef120d3f2578fe3d52eb6c0d0641df945991391 Ajax: make jQuery#load "type" field explicit
+ 97ef1f2612a9c5bd453d34146fdfd552cf9cee67 Ajax: replace "jqXHR.complete" callback with "always"
+ 8b6aeae52d4c53a53468678ccd45e9dda9665004 Build: fix broken assertions caused by QUnit update
+ 8356281bed643bb3d56ad02f52580a0e20dc0237 Tests: make top of the HTML suite compliant with style guide
+ a4715f4216ace92fba6991106053415e66289686 Ajax: remove use of jQuery#each second argument
+ 2380028ec4a6a77401b867a51de26a3cb8e8d311 Core: change jQuery.each and jQuery#each signatures
+ 95fb798980d7e404c413e29e20016db9052e2bf2 Data: avoid Object.defineProperties for nodes
+ d702b7637a61e1973e08c27b8d8de2ed24a543e2 Data: move element cache to element[expando]
+ 222ac3ad6bd40ef4dfb4e4c60bac4b751d907e2a Data: updates to element[expando] cache
+ 361a0d5150a1c57b1857611cde1b05bd0ef21a50 Event: correct support comment
+ 34da7d552982d8ab7b18c2ceca9786d5023930f6 Build: remove deprecated JSHint options
+ 9b03f6df88a8d9dbda3f7893cdd84e3a3c70da17 CSS: Support relative adjustment in any applicable unit
+ 332fd941b4ae80e8ca5e5a20aca91806038f4816 Data: restore explicit data removal of private data in cleanData.
+ aaeed53e9f1e299975cb6f697c8038ec3a83fa4d Attributes: return null when attribute does not exist
+ 5dc4616ca0fdbf2a80890bd9b236e796b84db8c1 Attributes: fix failing test for new return value
+ 7632b7454b8a918b5eb9ade199a6a842eae98571 Attributes: revert returning null for non-elements
+ 48be675200e817f40fa8ec25da1be2ab0839d28d Tests: Fix CSS relative adjustment test for round-down browsers
+ e22ef5d9017c44cad97ae541fefce76cc455edcb Tests: Expand CSS relative adjustment tolerance for IE
+ 62ae2d0fb7ac011bf2ad778f8158de408e785927 Offset: Round offset value for the sake of floating errors
+ 98c25b7c803fe708a587f70e5b70541192004247 Build: save sinon update for later
+ 0065e1f3e3021fc2bcb32e915177498bca034b34 Build: Update native-promise-only
+ 95c0a10e15477a5031185e2d656d896905562afa Data: Drop the tests relying on applets
+ bd9a1385feea349f48de658875c3f1bfc75daf74 Tests: add the current version of node and iojs to the travis config
+ f5aa89af7029ae6b9203c2d3e551a8554a0b4b89 Build: Update native-promise-only (again)
+ 06f6cd1ffd2a4e9f5955d146d229492f245d83cb Core: Update tested jsdom, drop obsolete workarounds
+ 2905961bf75b741937f430ee5b619f11e93a448d Attributes: revert returning null for non-existant attributes
+ 062b5267d0a3538f1f6dee3df16da536b73061ea Docs: Add info about Sizzle not being excludable on the compat branch
+ 3a0dd5a3d3a23e81dfb32af2871fab6be4619434 Build: update Sizzle
+ bb4d888f4f7886371347b59eae5d4e6135acb7ed CSS: Don't expose jQuery.swap
+ 892625b3c36eda6bb6ac226d15e0a158ba35cf21 Core: Follow the AMD specification for define
+ 55ac56aeda786a4d1b677aefd3f5bb134ecb02ad Attributes: remove unnecessary element null check
+ 002240a6eb1cee2fcd886d5cf44893eb67f246f1 Deferred: Always handle progress callbacks before done/fail
+ f6de5a90ddde29e9096c2e45b9da21e65882b7c0 Core: Align branches: remove an unused variable, add comments
+ ff18d8e2060ae7c15c7694dc6bcbbeb9cbfbdaa4 Tests: Increase QUnit timeout
+ ab407258795bfa05756b009058757f8b42aa9c53 Core: Test all factory use cases from intro.js
+ 154166458284bcce7d6a86328b7fd13483232a1a Core: add workaround for iOS JIT error in isArrayLike
+ 4b27ae16a2b911f75b341b56d9d939bc65a9657a Manipulation: Detect sneaky no-content replaceWith input
+ 225bde37c997f5ddd9fe00fdfb8e9a43545cfbbc Manipulation: Make an HTML interception point
+ 9db9316609c2881dbb6abc49efc3aa91a57a02ad Event: add support comment
+ a117dd05f638a078c21dc57f19966f4ae81f98f0 Ajax: remove "onunload" event handler
+ 85577a348a72ae765e0d7330b9e82985d23c94b7 Core:CSS:Event: simplification of native method signatures
+ a873558436383beea7a05fd07db7070a30420100 Event: remove redundant guards for the event methods
+ 7a6931de689865f559bed26e797e8cbb5674bbce Core: allow init to accept an alternate rootjQuery for migrate's sake
+ e1090c3d2b2a988a5b41f1f1ed9f8d6dcae02200 Selector: add jQuery.uniqueSort; deprecate jQuery.unique
+ add85afed5944ec10d68ca10e91421e031fe0a5d Selector: add test for jQuery.unique() alias
+ b92acf72372c3ad09d5f67ba3a1b2c19355a0e5f Build: Remove npm from dependencies
+ 436f0aee0a4c37ecb61b87a3a44227f6f5153b3a Core: remove isArraylike check for nodes
+ 5f2ea402582c7b8f4773771e1529d60587f3435e Effects: set default easing using jQuery.easing._default
+ 6d7ef56ed3004a18f5139928455cdbdc169e1232 Effects: add tests for jQuery.easing._default in Animation and Tween
+ a74320fca8bb0a2190f6e1fdb71a73733b6986e4 Manipulation: privatize buildFragment() function
+ 62d5579578109f1468a37e44f76af06f283a46ab Manipulation: privatize internal domManip() function
+ 5c3101fee60046fa7976b3131fada8dfe9fbd53e Build: use different versions of jsdom for Node and iojs testing
+ 7855a1a7d8a5c959d4ef2f951b4eb971908ac96f CSS: Collapse a double if statement into one
+ 79bcb291324167ab2d844027d0cdc6300613d010 CSS: fix :visible/:hidden selectors for inline element w/ content
+ 16713fb6097849727de2ba1b489c98e242a168c3 Tests: fix tests in accordance with new :visible behavior
+ 764dc949d0d65742606747ce75852d1b5dd59fcd Data: remove some unused code
+ 56bb677725b21415905e5c3eeb1e05be4480e780 Data: remove the expando when there's no more data
+ 20aaed367f993f3c2aa204183d82d0d73efa114f Attributes: add SVG class manipulation
+ 2d715940b9b6fdeed005cd006c8bf63951cf7fb2 Offset: account for scroll when calculating position
+ 61e21a4eaf479406b660328f4df7e3183c32386b Build: bower.json: remove moot `version` field
+ c074006a69db73a116dc04ec78844468a8cea7d3 Event: provide verbose comment for focus(in | out) & rename support prop
+ a644101ed04d0beacea864ce805e0c4f86ba1cd1 Build: update requirejs dependency to 2.1.17
+ 3c9277086742fe3a38a268ef97184be34e294655 Docs: remove redundant instruction from the readme
+ 7475d5debeb7c53158921ed40f6c2fdb25a2cc86 Event: Remove fake originalEvent from jQuery.Event.simulate
+ 6df669f0fb87cd9975a18bf6bbe3c3548afa4fee Event: remove outdated originalEvent hack
+ d471842b3e3af83c9a1be06b5d16f75bfa96af8c CSS: Don't cache unrecognized CSS property names
+ 8e92e1ea3c533f3be82c99bbafaaf74b5bdedecc Build: Update the license attribute
+ 0019a463bdcb81dc6ba3434505a45774ca27f363 CSS: Don't name the anonymous swap function
+ 3655c4e1908cc3ee49487e7e26e8cfca9fe8146d Event: remove preDispatch hook & simplify "simulate" signature
+ 285cfbfccc4c61d50ee8e0fe6e23695dc663e166 Build: remove bower.json lint target
+ 1556c4661af647e355a9a5c0a814012955e231bc Build: Update grunt-contrib-jshint
+ 9c8a3ecdc46156afd8f93aa44b6e6aea7c52c049 Build: Refactor Node smoke tests
+ bb026fc12c3c2ad37f47f0919e484bddcdc3d291 Core: Make jQuery objects iterable
+ 04a29696e5b176ac66401120e433d52425222f0f Event: Remove an internal argument to the on method
+ 349edbd6c53aa93d4fd207d3c0c4c24a7b0314dd Manipulation: Remove an internal argument to the remove method
+ c17543fd3c14ff86c448dbb90f9fe1223661a73b Tests: Correct a typo in the regex matching Safari 8
+ 8e111df641cca3e1b75b31a1971bfc89014b4ded Tests: Add Microsoft Edge results (from Windows 10 build 10130)
+ 5a1217e40193c8884155ccaf415091d326ddb52a Tests: Remove Edge version from the user agent
+ e831856490d2212bdbaff4cd76137b93ccf26d92 Tests: Remove test/data/ua.txt
+ 40dcc767640c41a4387a343f1ef53ac57ed631c5 Offset: return zeros for disconnected/hidden elements
+ 0e4477c676db0427bb9b0bf39df8631501e62f24 Offset: return before getBoundingClientRect to avoid error in IE8-11
+ b04124222395a05c80d4f1c3a70333fdb07bfe3d Offset: add tests for hidden elements + scroll
+ 219c7494938a10b985b7827990bc419e41585b10 Core: Use window.setTimeout & friends instead of global equivalents
+ 842958e7aecd0d75a7ee9e2aaec83457701aa2f3 Core: Switch from modules to just window.setTimeout etc.
+ 63a577aa0bcb439c1730c3825407d86c128b17be Build: space between curly and paren is optional
+ 90d828bad0d6d318d73d6cf6209d9dc7ac13878c CSS: Work around an IE11 fullscreen dimensions bug
+ 5153b5334eb2c8317372b46209bd9d092a91afdc Core: organize prop & attr code to be similar
+ 3a0d582cf63b6e8f77150d9c38d2bf34d0c7790e CSS: make the getStyles function more readable
+ cdaed15c7ea1bbfdde5a5bea691c583ce7961526 Effects: Add tests for jQuery.Tween
+ b3b2d6c3dd51fbdc69e1942e9af75cc99a1834c2 Effects: Adding unit tests for jQuery.Animation
+ a44cfa00665d21c3197e25c2d63741dc15f6ffb9 Tests: Lower the checks rounding error
+ 8e4aac8cb03ffb88373ea99629165d82ff5eccdd CSS: Improve a comment explaining IE11 fullscreen bug
+ 8887106702baa69ed80baa65c5a249786bffc77e CSS: Add an integration test for issue gh-1764
+ dc8ba6af921f4c9650fb1e4cfc698b2a276fa53d Tests: Remove a trailing comma for compatibility with the compat branch
+ 8f13997e89ca325a49f7581fad7e85fe37bad166 Build: update AUTHORS.txt
+ 3ec73efb26317239a4f22f0b023b0b99a4300a20 Build: add mailmap entry
+ a2ae215d999637e8d9d0906abcbf6b1ca35c8e6e Ajax: Remove jsonp callbacks through "jQuery#removeProp" method
+ c44dd7775b387094d8c921c7e839e3c266e4f2c8 Release: properly set the dist remote when it's a real release
+ bf591fb597a056bf2fc9bc474010374695b18d1a Selector: Define jQuery.uniqueSort in selector-native too
+ 1c59b308d201d6dd0f0aed2ad0256d01b9f68047 Build: Acknowledge Android 2.3 is not ES5-compatible
+ d24275372624bac897c4131fd1507a58c09a1483 Core: Add a support comment for Safari 8
+ bf48c21d225c31f0f9b5441d95f73615ca3dcfdb Core: .each/.map should accept an undefined/null value
+ 93bee4701d14202045a88aab156da0daf9418430 Core: Adjust comments & tests after dropping Safari 6 support
+ 2792845534e36c39dbb9c8369ed96aaefa560081 Event: Update support comments for mouseenter/mouseleave implementation
+ 6044fb6a7384aec85906949835ef9a58114896ce Ajax: Account for Android 2.3 not firing window.onerror on script errors
+ 5fe76c663f8a4986af62edb434a1708c006d0b21 Data: remove user data in cleanData
+ 360a4780339b7f412b75ad8a06dca7f39616f654 Docs: Fix various spelling mistakes
+ aabe94edb4880c75eeebc5b5b5d66a9ad17008fe Tests: don't use deprecated argument in test declaration
+ 669cb16d763cb5486dadd56ec15a17b2b0303571 Build: Add a comment explaining why the es3 option is needed
+ 9adfad19865837f5dffedb1eb41e407f196ca515 Core: Don't expose jQuery.access
+ 9d820fbde6d89bc7a06e2704be61cf6c0b4d6e3c Event: Only check elements for delegation matches
+ b930d14ce64937e9478405eee2828d4da091d2cb Tests: partially use new qunit interface
+ 7aa46e0df8a673e6b00550bbbbed21eed50108b7 Tests: do not define two modules with the same name
+ 10fdad742a2a6aa9f0e00b3e04fc5264797c53c7 Build: Update jscs and lint files
+ 99975c44abc8c1092764c7fca3ccfe89fc832b87 Build: correct jscs paths
+ e35bdc1a22d7b5786b8f849acdc4653f1dc25e9b Build: correct style tests files which could be automatically corrected
+ c9cf250daafe806818da1dd207a88a8e94a4ad16 Ajax: do not quote "throws" option - use dot notation instead
+ 835e9218beef8f0c559da314fac01cb85dacb740 Build: Update grunt-contrib-uglify because of a security issue in uglify
+ 3d7ce0a65f0707ff01a851822e57ba80adcff075 Serialize: Handle arrays with null values
+ 14c0fe44328f22debb5b531d2b671923658542b3 Serialize: Fix object detection
+ 02e10082b25df3b47b6b98e3b35399712795b7df Tests: lower the PHP sleep time in unreleasedXHR.html
+ 224271982eb9cd351d7db1b38c740b4e927e6f97 Data: Don't expose jQuery.acceptData
+ 5adf04a73c135dc729b9d9889bc963b45a9fc471 Build: put back "lint" command to the "dev" list
+ f5bf9bc48897e3b8f050d87d02252c8be456044a Data: do not create data cache when fetching single property
+ ab30934c1d7e023d9e80525c6e7ad4da31b543af Build: Fix an email address of a contributor
+ 98cee73244d55910a1ac82bcf6cae04a7f650484 Core: Remove unnecessary parameter to jQuery#constructor
+ 3dd3d1357d027a487559516fcdd9064cc792edab Effects: Finish should call progress
+ c8d15a2f9f108e90d3651c31e4abf45415a30fde Tests: further improvements QUnit 2.0 migration
+ c161eecce09fc66ce252d4adf64b565f726bb6d2 Tests: more style corrections
+ f9ef427d355a8a2b531aed5d112dfa4f9692105c Traversing: Don't expose jQuery.dir & jQuery.sibling
+ dbb2daa8c3a458e3525d310440f4365548edc138 Build: Don't install jsdom 3 on Node.js 0.10 & 0.12 by default
+ 2c7e9c934971500a746d012c529e13ec0b560a83 Tests: Add simple tests for Android 2.3
+ 855b0c8c288533948b257925a8906f7da3449eed Tests: Don't load non-basic tests when basic module is selected
+ 5914b103627e3773418ad1fd8c3b034bf3748d51 Tests: Make basic tests work in IE 8
+ 1b566d37a2879af12364a03c633c235a76f49925 Tests: Really fix tests in IE 8 this time
+ d4def22e4cd1c2eb2571f449e226b38384fb6d81 Manipulation: Switch rnoInnerhtml to a version more performant in IE
+ 0e982433eb94391b3e9f6838d9b8fbf9bb31abf9 Data: avoid using delete on DOM nodes
+ d3d8d9751f3d14a545b26cf820dc1f51896a7b50 Tests: Provide equal() arguments in correct order (actual, expected)
+ 250a1990baa571de60325ab2c52eabb399c4cf9e Build: Drop io.js testing, test on latest Node.js
+ 06320c88af5be9cd937917282deb6eb4c4ad0443 Build: Use double quotes in .travis.yml
+ cea94a83c7a5c5df6da578d417e5fb26152b19e1 Build: Remove a double empty line at the end of .travis.yml
+ 64fd7ef3d081b5c65d541237f73a4d89f0f0ad7b Tests: Account for Edge in originalEvent UA-sniffs
+ b5f7c9e2d526b17b9962976bb704dce8779d7362 Data: Combine register and cache methods
+ 99e8ff1baa7ae341e94bb89c3e84570c7c3ad9ea Manipulation: Don't provide the parser with sloppy table markup
+ f9af896bb8f4cb37b22d508443174c8edf40fc54 Tests: Add dummy modules when running basic tests
+ f23a64d745759d358d423ba0557e8f74d65f76cc Tests: Use QUnit URL parameter parsing
+ 735dea34fb0ae625542d51eae3f4e7316e403eaa Docs: "npm run build" was missing from the contributing guides
+ 9748e436ad80d6a2e1661ba4cf8d7391ed87c3ad Tests: Add .extend test for defined accessor properties
+ ee0854f85bd686b55757e8854a10480f23c928da Event: Move .bind() and .delegate() to deprecated
+ 03eaadb131df925d1072afd2496ee3b41d2f1fc6 Tests:Docs: Fix various typos
+ 0c34e688439713725d4215c63bc4cf876d8d0423 Build: Fixed issue with base path that contain 'var'
+ d92310050ca7bf0b33825d64e052f9a8809c3e9e Event: fix incorrect test
+ cb087ce41daa5db4c8db10e586bdc141f953d93d Ajax: correct indentation
+ 239169bb2ede6ea6287d82d1d13b0c354f451749 Ajax: improve content-type detection
+ 8a734344f2566dab5b8d34ecd79ae81ebd8843c5 Core: Support Symbol wrapper objects in jQuery.type
+ 625bd303d99408bca86b7002cd42b8716e2b267e Tests: fix code style issues
+ 67b76f5986a78d868553b9aec0d0192f5259a078 Tests: Account for array-like objects in jQuery.grep
+ 44f8239d3f9d09d81e2885e9ae7e044277b67341 Build: update Sizzle to 2.2.1
+ 0703fd52ef88a2cdac93502070c51c93ffa1dfdd Core: make isNumeric test work on Symbol
+ 0fb84fa8ccefcd07febf282fd7b80262ad70add7 Tests: Keep test iframes around for assertions
+ e271f665dd8fb617a5015051e1c9df8cebd6c97b Deprecated: fix amd mode for the deprecated module
+ 53f798cf4d783bb813b4d1ba97411bc752b275f3 Attributes: Remove undocumented .toggleClass( boolean ) signature
+ 4bf1a09522955eb52de1fafb4ee1ecc5982b7a3e Attributes: Use simpler boolean check vs a function call
+ 5db1e053098af747330044d5740e5379f2834402 Attributes: removeClass() -> attr("class", "")
+ c82a6685bb964627e27008e298f93ea81218265b Event: Fix delegated radio events when arrow keys are used
+ c577928d45bdcc3ee8f93da89ab7aadca21919f3 Tests: Change quotes according to style guidelines
+ 9f9e204bba41b7a9cde5ba7e065d817ef8b18c41 Offset: allow small differences in offset.top
+ 9090d98439f8dc449beafee98f8ff35cfb4f9116 Core: add unit test for isPlainObject(Symbol)
+ 3689963909880ed832ac17eabf7b9260927a68d8 CSS: use isFinite in place of redundant isNumeric
+ c752a5030bc00eb5b45dea9c28963f824a5c4f44 Attributes: fix tabIndex on <img> in IE11
+ 487d5ca913c237aafe9efa1179749b46382fddbf CSS: Correct misrepresentation of "auto" horizontal margins as 0
+ e99a3ac7066226d00ff5828de596d4a4b3818c0c Tests: Add Safari 9 support tests results
+ 1144e754a6a131bd4affec26fd85299e71bdab06 Build: Add "deprecated" to the Testswarm module list
+ 15ac848868e993dfe5ccd7751a94f5c8edc288bc Core: make isNumeric limited to strings and numbers
+ 5b554cf04e809d77026d7afba6f02a7599724377 Tests: Use standard external domain name
+ f931786018058174fa63551a7a4a3fccf9de41fa Tests: Post-Summit cleanup
+ dabd5ba96c05279b3ffb052db5b8d17f75996694 Core: use interactive to evaluate dom ready, barring IE9-10
+ 7bb39f35118c33f1f7effc49112579ec9393f02c Release: remove antiquated release-notes.js
+ cbe5b2b79a46c61ad36ea5df44f80b1cc2c1b60a Build: Test on Node 5
+ 1c2b5362ef40058d5b375a40362c88559f81a293 Tests: Add iOS 9 support tests results
+ 47e2aa6593a77c94eef610ac784494884f598253 Tests: Remove Safari 7.0 & iOS 6 support tests results
+ 015d16c02dae770eda88e644ec69ce82f25c0412 Tests: Make regexes for iOS devices more rigid
+ 76e9a95dbeaf28fbc5a64571ebb5959f91a9c14a Ajax: trigger error callback on native abort
+ 769446c69775f6c44e35cee1bcdeccafba51be7b Ajax: Don't throw exceptions on binary data response
+ 493b0fd7f51054345abe981270bd7839825f79dd Event: Ensure delegation doesn't error on comment nodes
+ e05c63e17a037d550e7dde5d805ee5c4214ee44b Event: Add basic unit tests for event aliases
+ 1da144867f3a49bbac4342bd3f3242baae108eb9 Tests: make editorconfig match css style
+ c9935b6d2db9e1be4bed12f7419e98cdca45763e Event: Separate trigger/simulate into its own module
+ 2cb8ebadcb77da1c1a12c07bc5274fd456bf3b01 Event: Make event aliases optional in unit tests
+ 67fa2eab6ef323b1d894e9e7f054c6e8c844d304 Event: Move VML test out of event alias test
+ 75b3cdd509a2cf0a704767d04557ec392112a438 Dimensions: properly manipulate non-px values
+ 29370190605ed5ddf5d0371c6ad886a4a4b5e0f9 Offset: offsetLeft/Top on empty set returns undefined
+ 88046440da8f5433b510ea705255d1df12c2963e Selector: pass jQuery unit tests with selector-native
+ 8a896dfac8ea9d7307d8819ab848fef9b03cd933 Ajax: code style fixes
+ c69673fe41ee17f46545e87a31ff96cea6c68a17 Release: remove revert artefacts
+ 3655260866cdcfe42a1117bfb9603144c9e4d829 Release: remove revert artefacts
+ e2af987877c83e5b054e1f9cac1a8534ebed0b18 Release: fix revert artefacts
+ f0532a29e38d01d6f3e44435198527323d507cf9 Callbacks: bring back size reduction
+ 9d1d4c272a58ced36242d90b3f0462c2bbb972a3 Offset: do not run tests which break without back-incompat change
+ 0a98623abb85bdce079b400ed3bf3d87ddc6b1da Effects: manually revert two `requestAnimationFrame` commits
+ ae88b3971c38e0d32a8b927d597426bb50263c6f Release: fix revert artefacts
+ 1ad9915d11e27ebce8016ef81163206fb68b2335 Misc: Fix the tests, revert some unneeded/broken reverts
+ c56e8b680dbfcbcdff77d982618a8c6eda68cb49 Build: Change the 2.2-stable version to 2.2.0-pre
+ f6dd76709a4173be5e8d45c3f901a330f751c382 Tests: Fix support tests results
+ 283a1949550e1bf1284ddf89a472b3c8b4a97c60 Tests: Blacklist the iframe scrollTop test in Opera 12.1x
+ e904249ee00c3375fad461c47150627fce0b02d8 Tests: Fix Safari 5.1 support tests results
+ a5864aefdfadfee16522456c82139fa10360d8fc Effects: Remove tests for jQuery.Animation & jQuery.Tween
+ 4fea389ad2b7515bc3a9081245979ab7be566b99 Tests: Fix support tests results for Android 2.3


This is what I noticed...

## Should be reverted

- isNumeric behavior change, checking type is string or number (15ac848868e993dfe5ccd7751a94f5c8edc288bc)

## Features that have been added. Some of these need to be documented/also make it into 1.12-stable.

- Symbol support (with for-of)
- jQuery.uniqueSort added as alias for jQuery.unique
- jQuery.sub fix (with root added back an argument to fn.init). This is for migrate.
- SVG class support
- Ajax: $.post and $.get can now take an options object

## Commits in 2.1.4 not in 2.2-stable

- e5190982c40d7ac8ab9bdb2e7e4334f0e123ef66 Ajax: Remove workaround for IE6/7
- 0cdec797de23555c95a70978f4d9e06f3b041330 Data: avoid non-alphanumeric chars in expando properties
- 4e7f34f6296111f7f91d621397dfb02c6bf4c41f Ajax: remove event dependency from the ajax module
- 7d15b4d1f181de8868c375c545a51e4dfc28a611 CSS: Clean up memory leak in reliableMarginRight

Of these, I think only the last one is important. We probably want that memory leak gone in 2.1.4.

**Comment by gibson042** [comment]
> ## Commits in 2.1.4 not in 2.2-stable
> 
> - e5190982c40d7ac8ab9bdb2e7e4334f0e123ef66 Ajax: Remove workaround for IE6/7
> - 0cdec797de23555c95a70978f4d9e06f3b041330 Data: avoid non-alphanumeric chars in expando properties
> - 4e7f34f6296111f7f91d621397dfb02c6bf4c41f Ajax: remove event dependency from the ajax module
> - 7d15b4d1f181de8868c375c545a51e4dfc28a611 CSS: Clean up memory leak in reliableMarginRight
> 
> Of these, I think only the last one is important. We probably want that memory leak gone in 2.1.4.

I believe we want all of them except the first. 0cdec797de23555c95a70978f4d9e06f3b041330 avoids a performance regression, and 4e7f34f6296111f7f91d621397dfb02c6bf4c41f is good hygeine.

**Comment by timmywil** [comment]
> ## Commits in 2.1.4 not in 2.2-stable
> 
> - e5190982c40d7ac8ab9bdb2e7e4334f0e123ef66 Ajax: Remove workaround for IE6/7
> - 0cdec797de23555c95a70978f4d9e06f3b041330 Data: avoid non-alphanumeric chars in expando properties
> - 4e7f34f6296111f7f91d621397dfb02c6bf4c41f Ajax: remove event dependency from the ajax module
> - 7d15b4d1f181de8868c375c545a51e4dfc28a611 CSS: Clean up memory leak in reliableMarginRight
> 
> Of these, I think only the last one is important. We probably want that memory leak gone in 2.1.4.

Actually, I take this back. They are all in 2.1-stable. The "-" doesn't mean not present, it means already picked. The "+" means not present in 2.1.4.

**Comment by markelog** [comment]
@gibson042 

> EDIT: And it might even make sense to include them all and then add a revert of e519098.

Why?

@timmywil 
> Features that have been added. Some of these need to be documented/also make it into 1.12-stable.

There is also `jQuery.htmlPrefilter` and `jQuery.Animate` API changes

@timmywil, @gibson042, @gnarf, @dmethvin 
Also need attention for "Questionable commits" and for "Complicated reverts" lists

**Comment by timmywil** [comment]
The complicated reverts look good to me. As far as questionable commits, all of the "Hides x" commits are fine. The one where gnarf was adding unit tests for Animation/Tween, I think we can keep those since that's the way it was supposed to work before. The other minor API additions, signature changes, and dropping of support for old browsers are also fine.

I think that's everything. Revert the isNumeric change, and make sure that 1.12-stable has the same API additions as 2.2-stable.

**Comment by mzgol** [comment]
> The one where gnarf was adding unit tests for Animation/Tween, I think we can keep those since that's the way it was supposed to work before.

I've already removed those tests, we talked about that. They were causing problems for Android 2.3 and we decided that even if they generally work it's not a problem we'll officially start supporting those use cases only in jQuery 3.0.

**Comment by timmywil** [comment]
>  They were causing problems for Android 2.3

Ah, right, that's fine too.

**Comment by timmywil** [comment]
To be specific, these are the API additions that need to also be in 1.12-stable. If there's any I missed, I'll add them to this comment.

- Symbol support (with for-of)
- jQuery.uniqueSort added as alias for jQuery.unique
- jQuery.sub fix (with root added back an argument to fn.init). This is for migrate.
- SVG class support
- Ajax: $.post and $.get can now take an options object
- `jQuery.htmlPrefilter`

**Comment by markelog** [comment]
Coolsies! `isNumeric` changes are reverted, will start working on 1.12 tomorrow.

From that point on, i'm asking all interested parties to be very certain in adding/reverting commits for those versions.

I'm planning to do most of the work for the next meeting and release in middle of the next week. Hope that sounds good to everyone. If not, please express your concerns.

Also, would we like to document API changes? Or wait until 3.0.

**Comment by timmywil** [comment]
> Also, would we like to document API changes? Or wait until 3.0?

Yes, all API additions/updates will need to be documented as 1.12/2.2 additions. Specifically, this includes...

- jQuery.uniqueSort
- jQuery.htmlPrefilter
- $.post and $.get taking an object of options
- SVG class support – this can be a short note on the class docs

and **excludes**...

- Symbol support. I'm not sure how we'd document this or even if we need to, but this can wait until 3.0 regardless.
- jQuery.sub fix (init argument addition). This was purely for migrate.
